src/ucp/tag/offload/rndv.c

```diff
@@ -0,0 +1,309 @@
+/**
+ * Copyright (C) 2022, NVIDIA CORPORATION & AFFILIATES. ALL RIGHTS RESERVED.
+ * See file LICENSE for terms.
+ */
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include <ucp/proto/proto_single.inl>
+#include <ucp/rndv/proto_rndv.inl>
+#include <ucp/tag/offload.h>
+#include <ucp/tag/tag_rndv.h>
+
+
+static UCS_F_ALWAYS_INLINE int
+ucp_proto_tag_check_op_id(const ucp_proto_init_params_t *init_params)
+{
+    return (init_params->select_param->op_id == UCP_OP_ID_TAG_SEND) ||
+           (init_params->select_param->op_id == UCP_OP_ID_TAG_SEND_SYNC);
+}
+
+static UCS_F_ALWAYS_INLINE int
+ucp_proto_tag_offload_rndv_supported(const ucp_proto_init_params_t *init_params,
+                                     ucp_proto_rndv_ctrl_init_params_t *params)
+{
+    ucp_worker_h worker   = init_params->worker;
```

Comment:
Could be replaced by
```suggestion
    ucp_context_h context = init_params->worker->context;
```

Reply:
`worker` is also used in this func

Reply:
line 35:
```C
md_attr = &context->tl_mds[md_index].attr;
```

line 40:
```C
(context->config.ext.tm_sw_rndv == UCS_YES)) {
```